buildSrc/src/main/groovy/org/elasticsearch/gradle/test/StandaloneRestTestPlugin.java

```diff
@@ -0,0 +1,109 @@
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().withPlugin("elasticsearch.build", appliedPlugin -> {
+            throw new InvalidUserDataException(
+                "elasticsearch.standalone-test "
+                    + "elasticsearch.standalone-rest-test, "
+                    + "and elasticsearch.build "
+                    + "are mutually exclusive"
+            );
+        });
+
+        project.getRootProject().getPluginManager().apply(GlobalBuildInfoPlugin.class);
+        project.getPluginManager().apply(JavaBasePlugin.class);
+        project.getPluginManager().apply(TestClustersPlugin.class);
+        project.getPluginManager().apply(RepositoriesSetupPlugin.class);
+        project.getPluginManager().apply(RestTestBasePlugin.class);
+
+        project.getTasks().register("buildResources", ExportElasticsearchBuildResourcesTask.class);
+        ElasticsearchJavaPlugin.configureInputNormalization(project);
+        ElasticsearchJavaPlugin.configureCompile(project);
+
+        project.getExtensions().getByType(JavaPluginExtension.class).setSourceCompatibility(BuildParams.getMinimumRuntimeVersion());
+        project.getExtensions().getByType(JavaPluginExtension.class).setTargetCompatibility(BuildParams.getMinimumRuntimeVersion());
+
+        // only setup tests to build
+        SourceSetContainer sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
+        final SourceSet testSourceSet = sourceSets.create("test");
+
+        project.getTasks().withType(Test.class).configureEach(test -> {
+            test.setTestClassesDirs(testSourceSet.getOutput().getClassesDirs());
+            test.setClasspath(testSourceSet.getRuntimeClasspath());
+        });
+
+        // create a compileOnly configuration as others might expect it
+        project.getConfigurations().create("compileOnly");
+        project.getDependencies().add("testImplementation", project.project(":test:framework"));
+        EclipseModel eclipse = project.getExtensions().getByType(EclipseModel.class);
+        eclipse.getClasspath().setSourceSets(Arrays.asList(testSourceSet));
+        List<Configuration> plusConfigurations = Arrays.asList(
+            project.getConfigurations().getByName(JavaPlugin.TEST_RUNTIME_CLASSPATH_CONFIGURATION_NAME)
+        );
+        eclipse.getClasspath().setPlusConfigurations(plusConfigurations);
+        IdeaModel idea = project.getExtensions().getByType(IdeaModel.class);
+        idea.getModule()
+            .setTestSourceDirs(
+                Stream.of(idea.getModule().getTestSourceDirs(), testSourceSet.getJava().getSrcDirs())
+                    .flatMap(x -> x.stream())
+                    .collect(Collectors.toSet())
+            );
+        idea.getModule().getScopes().put("TEST", Map.of("plus", plusConfigurations));
+
+        BuildParams.withInternalBuild(() -> InternalPrecommitTasks.create(project, false)).orElse(() -> PrecommitTasks.create(project));
```

Comment:
the actual change in this class except porting it to java

Reply:
Since we removed jarhell and thirdpartyaudit from `InternalPrecommitTasks` what then applies those? Shouldn't we _always_ apply `PrecommitTasks`, not just as part of this `else` condition?

Reply:
`PrecommitTasks` are applied within `InternalPrecommitTasks`